### PR TITLE
room capacity limits

### LINKED ISSUE #89: add room capacity limits
20 for free users, 100 for subs?

Currently 100 is the limit for vbrowsers due to UDP port assignment. This can be increased if needed but we will run out of bandwidth at some point.

### PR DESCRIPTION
**Pull request by Argn0**
closes #89

### CODE REVIEW

File: server/room.ts
Comment (howardchung) [inline server/room.ts:44]: probably should just default this to ROOM_CAPACITY

File: server/room.ts
Comment (howardchung) [inline server/room.ts:70]: I would rename this to just "isRoomOwnedByCurrentSubscriber" and then do the side effect directly in this constructor

File: server/room.ts
Comment (howardchung) [inline server/room.ts:113]: Do we need to broadcast the room capacity?

File: server/room.ts
Comment (howardchung) [inline server/room.ts:145]: can't we tie this into the setRoomState function?

File: server/room.ts
Comment (howardchung) [inline server/room.ts:161]: I'd actually be fine with rooms retaining their larger capacity even after the owner's sub expires (they just can't make any new large rooms). This also aligns it with the current behavior of vanity names. If we decide we want to clean them up later we can do both those things in a cleanup function.

That might skip the need for checking the sub on each room when it's rebuilt from redis (which may add to startup time as we're hitting the database thousands of times)

File: src/components/App/App.tsx
Comment (howardchung) [inline src/components/App/App.tsx:531]: yeah rather than networking the room capacity value let's just make these react app accessible config values? (prefix with REACT_APP)

File: src/components/Modal/SubscribeModal.tsx
Comment (howardchung) [inline src/components/Modal/SubscribeModal.tsx:102]: Max Room Capacity?

File: src/components/Settings/SettingsTab.tsx
Comment (howardchung) [inline src/components/Settings/SettingsTab.tsx:131]: I'd move "Permanent rooms of subscribers have a greater room capacity." after the "Permanent rooms aren't deleted and can have passwords/custom URLs."

File: src/components/App/App.tsx
Comment (Argn0) [reply src/components/App/App.tsx:531]: If we didn't broadcast the room capacity, wouldn't we have to broadcast if it's a sub room or not instead?

File: server/room.ts
Comment (Argn0) [reply server/room.ts:161]: I thought about adding a new column "isSubRoom" to the table for this. Does that sound fine?

Comment (howardchung) [reply server/room.ts:161]: Yeah, I think that'd be fine. We just want to avoid the performance penalty of having to check sub state for thousands of rooms on boot.

File: src/components/App/App.tsx
Comment (howardchung) [reply src/components/App/App.tsx:531]: Actually after looking at this again this is only needed for the display on the member list right? If we don't want/need to display the max then we don't need to send it to the client at all.

**Comment by Argn0** [comment]
So I reduced the scope of this PR.

File: server/room.ts
Comment (Argn0) [reply server/room.ts]: I will try to solve the  "upgrade" problem after you have approved the current changes

File: server/room.ts
Comment (howardchung) [inline server/room.ts:82]: should probably match cases here so it's not confusing. Use camel_case if needed?

**Comment by howardchung** [comment]
Where's the logic for setting isSubRoom?

**Comment by atilla4455** [comment]
> Where's the logic for setting isSubRoom?

line 689 in room.ts

**Review by howardchung — APPROVED** [review]

File: server/room.ts
Comment (howardchung) [reply server/room.ts]: this looks fine to me. However, the upgrade problem is going to require one of the following:
* Keep a copy of subscribers in our local db
* have stripe notify us somehow when a new subscription happens (but this notification might be lost so we need a backup to make this consistent)

Comment (Argn0) [reply server/room.ts]: How about stripe webhook notifications?

Comment (Argn0) [reply server/room.ts]: > If Stripe does not receive a 2xx HTTP status code, the notification attempt is repeated.

Comment (Argn0) [reply server/room.ts]: Or maybe we could just upgrade the room(s) when the subscriber logs in and ask people to relog into their accounts if they want their rooms to be upgraded.

Comment (Argn0) [reply server/room.ts]: Actually no, forget that...

**Comment by howardchung** [comment]
This PR needs to be updated to use the new double quoted columns right?

We may need to solve the upgrade problem before we can ship it though

**Comment by howardchung** [comment]
note that I made the db schema update in a separate commit, you can rebase to fix the conflict

**Comment by howardchung** [comment]
I think this is mergeable, once rebased and conflicts fixed

**Comment by howardchung** [comment]
Also, there's a FAQ question to update about room capacity

**Comment by Argn0** [comment]
Don't merge yet please.

**Comment by howardchung** [comment]
oh I forgot the upgrade problem is still a blocker here. We could ship the code and I could set the limits to 0 for now until we figure it out

**Comment by Argn0** [comment]
How about we add a setting for subs to adjust room capacity, so people can increase it themselves?

**Comment by Argn0** [comment]
This is such a pesky issue...

**Comment by howardchung** [comment]
Hmm that's not a bad idea, but I feel like we want the seamless experience where the user doesn't have to manually change it on their rooms

**Comment by Argn0** [comment]
OK, I'm done with testing these changes. You can merge this if you'd like to.

**Pull request merged**
